### PR TITLE
Replaces Object.assign with ... spread

### PR DESCRIPTION
**Pull request by lukaszsobek**
Associated Issue: #4568

### Summary of Changes

* replaced all instances of Object.assign with ... spread

### Test Plan
a lot of "yarn test"

### CODE REVIEW

**Comment by bomsy** [comment]
Nice! Some lint and flow issues.
image

image

**Comment by jasonLaster** [comment]
hmm, looks like you only affected one test, which is a good thing `editor-select`.

You can take a look at our mochitest docs, or try to hunt it down

File: src/components/SecondaryPanes/EventListeners.js [hunk 1]
```diff
@@ -97,14 +97,15 @@ class EventListeners extends Component<Props> {
 
 export default connect(
   state => {
-    const listeners = getEventListeners(state).map(l =>
-      Object.assign({}, l, {
+    const listeners = getEventListeners(state).map(l => {
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/EventListeners.js]: ` getEventListeners(state).map(l => ({})` will work too

File: src/test/mochitest/examples/long.js [hunk 2]
```diff
@@ -5,66 +5,68 @@ var app = {};
...
 
   this.inform();
 };
 
-app.TodoModel.prototype.clearCompleted = function () {
-  this.todos = this.todos.filter(function (todo) {
-	return !todo.completed;
+app.TodoModel.prototype.clearCompleted = function() {
+  this.todos = this.todos.filter(function(todo) {
+    return !todo.completed;
```
Comment (jasonLaster) [inline src/test/mochitest/examples/long.js]: lol this might be it... revert these changes as it's a file that is tested.

it'll probably fix the test

File: src/components/SecondaryPanes/EventListeners.js (hunk 1 above)
Comment (lukaszsobek) [reply src/components/SecondaryPanes/EventListeners.js]: I was actually perplexed for a while as how to do it, thanks for the follow up!

File: src/test/mochitest/examples/long.js (hunk 2 above)
Comment (lukaszsobek) [reply src/test/mochitest/examples/long.js]: The fun thing is that yarn test runs 100% clean.

Comment (jasonLaster) [reply src/test/mochitest/examples/long.js]: yep... different suite. great work by the way

Comment (jasonLaster) [reply src/test/mochitest/examples/long.js]: it is so readable

Comment (lukaszsobek) [reply src/test/mochitest/examples/long.js]: Thank you, this issue was 250% worth the time spent.

**Comment by jasonLaster** [comment]
just did a rebase

**Comment by jasonLaster** [comment]
two failures:

TEST-START browser_dbg-chrome-create.js
TEST-START browser_dbg-editor-select.js

**Comment by lukaszsobek** [comment]
Not really sure what to do here...
